app/common/src/utilities/data/types.ts

```diff
@@ -0,0 +1,49 @@
+import { isRef, type Ref } from 'vue'
+
+/**
+ * A value that is not an {@link Array}, or {@link object} unless it is a {@link Function} or Vue
+ * {@link Ref}.
+ */
+export type ScalarOrRef =
+  | undefined
+  | number
+  | string
+  | boolean
+  | bigint
+  | symbol
+  | null
```

Comment:
In JS world this set of types is called rather ["primitive"](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) than "scalar".